code/modules/mob/living/taste.dm

```diff
@@ -4,21 +4,31 @@
 	var/last_taste_time
 	var/last_taste_text
 
+/*
+* Gets the "taste_sensitivity" of a given mob. This is used in calculating
+* what flavours the mob can pick up, with a lower number closer to 0
+* being better.
+*/
 /mob/living/proc/get_taste_sensitivity()
 	return DEFAULT_TASTE_SENSITIVITY
 
 /mob/living/carbon/get_taste_sensitivity()
 	var/obj/item/organ/tongue/tongue = getorganslot(ORGAN_SLOT_TONGUE)
-	if(istype(tongue) && !HAS_TRAIT(src, TRAIT_AGEUSIA))
+	if(istype(tongue))
 		. = tongue.taste_sensitivity
 	else
-		. = 101 // can't taste anything without a tongue
+		// carbons without tongues normally have TRAIT_AGEUSIA but sensible fallback
+		. = DEFAULT_TASTE_SENSITIVITY
 
 // non destructively tastes a reagent container
 /mob/living/proc/taste(datum/reagents/from)
+	if(HAS_TRAIT(src, TRAIT_AGEUSIA))
+		return
+
+	var/taste_sensitivity = get_taste_sensitivity()
+
 	if(last_taste_time + 50 < world.time)
-		var/taste_sensitivity = get_taste_sensitivity()
-		var/text_output = from.generate_taste_message(taste_sensitivity,src)
+		var/text_output = from.generate_taste_message(src, taste_sensitivity)
```

Comment:
Why call this before the cooldown?

Reply:
Do you mean why call `get_taste_sensitivity()` before the cooldown check?

Reply:
yes